design/BufferOperations.md

```diff
@@ -181,3 +148,111 @@ stagingVertexBuffer.mapWriteAsync().then((stagingData) => {
     // Enqueue copy from the staging buffer to the real vertex buffer.
 });
 ```
+
+### Updating data to an existing buffer (like webgl.bufferSubData)
+
+```
+function bufferSubData(device, destBuffer, destOffset, srcArrayBuffer) {
+    const byteCount = srcArrayBuffer.byteLength;
+    const (srcBuffer, mapping) = device.createBufferMapped({
+        size: byteCount,
+        usage: GPUBufferUsage.MAP_WRITE | GPUBufferUsage.TRANSFER_SRC,
```

Comment:
we should probably make `MAP_WRITE` implicit in this case. There is no use case for creating a mapped buffer for reading, or without the flag.

Reply:
If I remember correctly, the buffer descriptor for createBufferMapped does not need to have a map usage.

Reply:
Does it need to have TRANSFER_DST?

Reply:
No, we are only transferring *from* it into the destination buffer

Reply:
I mean, instead of MAP_WRITE. It must require one of them (either explicitly or implicitly) in order to put data in it.

Reply:
Never mind, I think I misunderstood.

IMO, createBufferMapped should not require any usage for the initial map, but it should also allow an explicit MAP_WRITE in case you want to map the buffer later.

Reply:
If you'd like to make that change, please file an issue!

Reply:
(This example doesn't MAP_WRITE later, but the other one does.)